tests/evaluate/test_default_evaluator.py

```diff
@@ -4128,3 +4134,84 @@ def word_count_eval(predictions):
             data["text"],
             check_names=False,
         )
+
+
+def test_do_not_log_built_in_metrics_as_artifacts():
+    with mlflow.start_run() as run:
+        model_info = mlflow.pyfunc.log_model(
+            artifact_path="model", python_model=language_model, input_example=["a"]
+        )
+        data = pd.DataFrame(
+            {
+                "inputs": ["words random", "This is a sentence."],
+                "ground_truth": ["words random", "This is a sentence."],
+            }
+        )
+        evaluate(
+            model_info.model_uri,
+            data,
+            targets="ground_truth",
+            predictions="answer",
+            model_type="question-answering",
+            evaluators="default",
+            extra_metrics=[
+                toxicity(),
+                flesch_kincaid_grade_level(),
+            ],
+        )
+        client = mlflow.MlflowClient()
+        artifacts = [a.path for a in client.list_artifacts(run.info.run_id)]
+        assert _GENAI_CUSTOM_METRICS_FILE_NAME not in artifacts
+
+
+def test_log_llm_custom_metrics_as_artifacts():
+    with mlflow.start_run() as run:
+        model_info = mlflow.pyfunc.log_model(
+            artifact_path="model", python_model=language_model, input_example=["a"]
+        )
+        data = pd.DataFrame(
+            {
+                "inputs": ["words random", "This is a sentence."],
+                "ground_truth": ["words random", "This is a sentence."],
+            }
+        )
+        example = EvaluationExample(
+            input="What is MLflow?",
+            output="MLflow is an open-source platform for managing machine learning workflows.",
+            score=4,
+            justification="test",
+            grading_context={"targets": "test"},
+        )
+        # This simulates the code path for metrics created from make_genai_metric
+        answer_similarity_metric = answer_similarity(model="openai:/gpt-4", examples=[example])
+        another_custom_metric = make_genai_metric_from_prompt(
+            name="custom llm judge",
+            judge_prompt="This is a custom judge prompt.",
+            greater_is_better=False,
+            parameters={"temperature": 0.0},
+        )
+        result = evaluate(
+            model_info.model_uri,
+            data,
+            targets="ground_truth",
+            predictions="answer",
+            model_type="question-answering",
+            evaluators="default",
+            extra_metrics=[
+                answer_similarity_metric,
+                another_custom_metric,
+            ],
+        )
+
+    client = mlflow.MlflowClient()
+    artifacts = [a.path for a in client.list_artifacts(run.info.run_id)]
+    assert _GENAI_CUSTOM_METRICS_FILE_NAME in artifacts
+
+    table = result.tables[_GENAI_CUSTOM_METRICS_FILE_NAME.split(".", 1)[0]]
```

Comment:
rsplit?

Reply:
TIL rsplit, but will use os.path.splitext that @B-Step62 suggested below as it's more catered to file path split.

thanks for the suggestion!

Reply:
I'd use `os.path.splitext`.

Reply:
In this particular code, I'm fine with `split`.